exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/InsertTests.kt

```diff
@@ -636,34 +638,45 @@ class InsertTests : DatabaseTestsBase() {
     fun testInsertIntoNullableGeneratedColumn() {
         val generatedTable = object : IntIdTable("generated_table") {
             val amount = integer("amount").nullable()
-            val computedAmount = integer("computed_amount").nullable().databaseGenerated()
+            var computedAmount = integer("computed_amount").nullable().databaseGenerated()
+
+            // Exposed does not currently support creation of generated columns
+            fun initComputedColumn() {
+                (columns as MutableList<Column<*>>).remove(computedAmount)
+                computedAmount = integer("computed_amount").nullable().databaseGenerated().apply {
+                    when (currentDialectTest) {
+                        is OracleDialect, is H2Dialect -> withDefinition("GENERATED ALWAYS AS (AMOUNT + 1)")
+                        else -> withDefinition("GENERATED ALWAYS AS (AMOUNT + 1) STORED")
```

Comment:
There are some limitations.
It would be great in this use case to do something like `.withDefinition("GENERATED ALWAYS AS", amount.plus(1))`.
But this does not work because Exposed generates expression SQL using the column's full identity `generated_table.amount`, which is valid for queries and other statements, but not valid syntax in a CREATE TABLE ddl.
So available expressions are those that do not use columns of the table being created at the time.